From be9f72bc5fa05de466f9151ac399c17027b58b24 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Wed, 12 Aug 2020 21:38:51 +0200
Subject: [PATCH] src/omx_reference_resource_manager.{h,c}: fix build with gcc
 10

Move globalIndex, listOfcomponentRegistered, globalComponentList and
globalWaitingComponentList from header to source file to fix the build
with gcc 10 which default to -fno-common:
https://gcc.gnu.org/gcc-10/porting_to.html

Fixes:
 - http://autobuild.buildroot.org/results/0c02f845eabfbc276ee5989520a3c77fc47289ab

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://sourceforge.net/p/omxil/bugs/63]
---
 src/omx_reference_resource_manager.c | 5 +++++
 src/omx_reference_resource_manager.h | 6 ------
 2 files changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/omx_reference_resource_manager.c b/src/omx_reference_resource_manager.c
index ee7766b..5983611 100644
--- a/src/omx_reference_resource_manager.c
+++ b/src/omx_reference_resource_manager.c
@@ -30,6 +30,11 @@
 #include "base/omx_base_component.h"
 #include "queue.h"
 
+int globalIndex;
+NameIndexType *listOfcomponentRegistered;
+ComponentListType **globalComponentList;
+ComponentListType **globalWaitingComponentList;
+
 /**
  * This is the static base pointer of the list
  */
diff --git a/src/omx_reference_resource_manager.h b/src/omx_reference_resource_manager.h
index ba8dad9..75e4f93 100644
--- a/src/omx_reference_resource_manager.h
+++ b/src/omx_reference_resource_manager.h
@@ -48,12 +48,6 @@ struct NameIndexType {
 	int max_components;
 };
 
-
-int globalIndex;
-NameIndexType *listOfcomponentRegistered;
-ComponentListType **globalComponentList;
-ComponentListType **globalWaitingComponentList;
-
 OMX_ERRORTYPE RM_RegisterComponent(char *name, int max_components);
 OMX_ERRORTYPE addElemToList(ComponentListType **list, OMX_COMPONENTTYPE *openmaxStandComp, int index, OMX_BOOL bIsWaiting);
 OMX_ERRORTYPE removeElemFromList(ComponentListType **list, OMX_COMPONENTTYPE *openmaxStandComp);
-- 
2.27.0

